controller/monitor/environment_check_monitor.go

```diff
@@ -233,36 +235,27 @@ func (m *EnvironmentCheckMonitor) syncPackagesInstalled(kubeNode *corev1.Node, n
 		return
 	}
 
-	notFoundPkgs := []string{}
-	for _, pkg := range packages {
-		args := options
-		if !pipeFlag {
-			args = append(args, pkg)
-		}
-		queryResult, err := nsexec.Execute(nil, queryPackagesCmd, args, lhtypes.ExecuteDefaultTimeout)
-		if err != nil {
+	installedPackages := []string{}
+	notInstalledPackages := []string{}
+	for pkg, executable := range packageProbeExecutables {
+		shellCmd := fmt.Sprintf("command -v %s", executable)
+		if _, err := nsexec.Execute(nil, "sh", []string{"-c", shellCmd}, lhtypes.ExecuteDefaultTimeout); err != nil {
```

Comment:
`command` is a shell function, not an independent executable, hence we have to wrap with `sh -c "..."`. The reason we don't use `which` is to minimize dependency. `which` is not installed on some Linux distro by default.